spec/controllers/catalog_controller_spec.rb

```diff
@@ -353,13 +353,19 @@
       end
 
       it "uploads a selected png file " do
-        file = fixture_file_upload('files/upload_image.png', 'image/png')
+        file = fixture_file_upload('spec/fixtures/image.png')
         post :st_upload_image, :params => { :format => :js, :id => @st.id, :upload => {:image => file}, :active_tree => :sandt_tree, :commit => 'Upload' }
-        expect(assigns(:flash_array).first[:message]).to include('Custom Image file "upload_image.png" successfully uploaded')
+        expect(assigns(:flash_array).first[:message]).to include('Custom Image file "image.png" successfully uploaded')
+      end
+
+      it "uploads a selected jpg file " do
+        file = fixture_file_upload('spec/fixtures/image.jpg')
+        post :st_upload_image, :params => { :format => :js, :id => @st.id, :upload => {:image => file}, :active_tree => :sandt_tree, :commit => 'Upload' }
+        expect(assigns(:flash_array).first[:message]).to include('Custom Image file "image.jpg" successfully uploaded')
       end
 
       it "displays an error when the selected file is not a png file or .jpg " do
-        file = fixture_file_upload('files/upload_image.txt', 'image/png')
+        file = fixture_file_upload('spec/fixtures/test.txt.png')
```

Comment:
If I recall correctly, this was intentionally a `.txt` file and not a `.png` file (regardless of the guts of the file) to demonstrate neither .png nor .jpg.

Reply:
Your comment makes a lot of sense.

The upload_spec.rb uses the name `test.txt.png` to show that the controller is validating the file contents and not the extension. Before #8934 `test.txt.png` incorrectly would work. Using that filename shows that we are validating the contents and not the filename.

This PR adds similar validation, so I thought it was a good convention.